src/helpers.cc

```diff
@@ -312,10 +296,17 @@ LiteClient::LiteClient(Config& config_in)
   KeyManager keys(storage, config.keymanagerConfig());
   keys.copyCertsToCurl(*http_client);
 
-  registerComposeAppManager(ostree_sysroot);
-
-  primary =
-      std::make_shared<SotaUptaneClient>(config, storage, http_client, nullptr, primary_ecu.first, primary_ecu.second);
+  // TODO: consider improving this factory method
+  if (config.pacman.type == ComposeAppManager::Name) {
+    package_manager_ =
+        std::make_shared<ComposeAppManager>(config.pacman, config.bootloader, storage, http_client, ostree_sysroot);
+  } else if (config.pacman.type == PACKAGE_MANAGER_OSTREEDOCKERAPP) {
+    package_manager_ = std::make_shared<DockerAppManager>(config.pacman, config.bootloader, storage, http_client);
+  } else if (config.pacman.type == PACKAGE_MANAGER_OSTREE) {
+    package_manager_ = std::make_shared<OstreeManager>(config.pacman, config.bootloader, storage, http_client);
+  } else {
+    throw std::runtime_error("Unsupported package manager type: " + config.pacman.type);
+  }
```

Comment:
This is turning into a pretty big function. I wonder if we should break this piece out into its own "mkpkg_mgr" helper? Might make adding unit tests for this specific thing easier if we ever got to it.

Reply:
If we remove PACKAGE_MANAGER_OSTREEDOCKERAPP in the future it will be just two 'if's.
Also, I am thinking about having just one type of package manager in the future just PACKAGE_MANAGER_OSTREE and introduce another component((s) responsible specifically for the app installation and a new composite component that would compose or aggregate the aforementioned objects (prefer composition over inheritance :)).

Reply:
sounds good.